### ISSUE TITLE
Online Batch in cluster mode with Having generates empty results sometimes

### ISSUE THREAD

**Issue by zhanghaohit** [4paradigm/OpenMLDB#2661]
**Bug Description**

```
  - id: 2
    desc: Group By And Having, Group命中索引
    mode: request-unsupport
    sql: |
      SELECT col2, sum(col1) as col1_sum, sum(col3) as col3_sum,
      sum(col4) as col4_sum, sum(col2) as col2_sum,
      sum(col5) as col5_sum FROM {0} Group By {0}.col2 Having sum(col1) > 10.0;
    inputs:
      -
        schema: col0:string, col1:int32, col2:int16, col3:float, col4:double, col5:int64, col6:string
        index: index2:col2:col5
        data: |
          0, 1, 5, 1.1, 11.1, 1, 1
          0, 2, 5, 2.2, 22.2, 2, 22
          1, 3, 55, 3.3, 33.3, 1, 333
          1, 4, 55, 4.4, 44.4, 2, 4444
          2, 5, 55, 5.5, 55.5, 3, aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
          2, 6, 555, 6.6, 66.6, 4, bbb
          2, 7, 555, 7.7, 77.7, 5, cccc
    expect:
      columns: ["col2 int16", "col1_sum int32", "col3_sum float", "col4_sum double", "col2_sum int16", "col5_sum int64"]
      order: col2
      rows:
        - [55, 12, 13.2, 133.2, 165, 6]
        - [555, 13, 14.3, 144.3, 1110, 9]
```

**Steps to Reproduce**

1. sql_cluster_test, make the tablet number to 2 or 3, it may generate empty results.
2. 
3. 
4.

**Comment by aceforeverd** [comment]
does it error with `toydb_run_engine` too ?

**Comment by zhanghaohit** [comment]
> does it error with `toydb_run_engine` too ?

no error in `toydb_run_engine`. it only failed in cluster mode with multiple tablets.

**Comment by zhanghaohit** [comment]
```c++
const codec::Row *next_row_iterator(int8_t *input) {
    ::hybridse::codec::IteratorRef *iter_ref =
        (::hybridse::codec::IteratorRef *)(input);
    ConstIterator<uint64_t, codec::Row> *iter =
        (ConstIterator<uint64_t, codec::Row> *)(iter_ref->iterator);
    auto res = &(iter->GetValue());
    iter->Next();
    return res;
}
```

after `iter->Next()`, the res is invalid in `RemoteWindowIterator`

**Comment by aceforeverd** [comment]
fixed by #2620
